nav2_costmap_2d/plugins/obstacle_layer.cpp

```diff
@@ -72,64 +77,62 @@ void ObstacleLayer::onInitialize()
 
   global_frame_ = layered_costmap_->getGlobalFrameID();
   double transform_tolerance;
-  nh.param("transform_tolerance", transform_tolerance, 0.2);
+  node_->get_parameter_or<double>("transform_tolerance", transform_tolerance, 0.2);
 
-  std::string topics_string;
   // get the topics that we'll subscribe to from the parameter server
-  nh.param("observation_sources", topics_string, std::string(""));
-  ROS_INFO("    Subscribed to Topics: %s", topics_string.c_str());
+  std::string topics_string;
+  node_->get_parameter_or<std::string>("observation_sources", topics_string, std::string(""));
+
+  RCLCPP_INFO(node_->get_logger(), "Subscribed to Topics: %s", topics_string.c_str());
 
   // now we need to split the topics based on whitespace which we can use a stringstream for
   std::stringstream ss(topics_string);
 
   std::string source;
   while (ss >> source) {
-    ros::NodeHandle source_node(nh, source);
-
     // get the parameters for the specific topic
     double observation_keep_time, expected_update_rate, min_obstacle_height, max_obstacle_height;
     std::string topic, sensor_frame, data_type;
     bool inf_is_valid, clearing, marking;
 
-    source_node.param("topic", topic, source);
-    source_node.param("sensor_frame", sensor_frame, std::string(""));
-    source_node.param("observation_persistence", observation_keep_time, 0.0);
-    source_node.param("expected_update_rate", expected_update_rate, 0.0);
-    source_node.param("data_type", data_type, std::string("PointCloud"));
-    source_node.param("min_obstacle_height", min_obstacle_height, 0.0);
-    source_node.param("max_obstacle_height", max_obstacle_height, 2.0);
-    source_node.param("inf_is_valid", inf_is_valid, false);
-    source_node.param("clearing", clearing, false);
-    source_node.param("marking", marking, true);
-
-    if (!(data_type == "PointCloud2" || data_type == "PointCloud" || data_type == "LaserScan")) {
-      ROS_FATAL("Only topics that use point clouds or laser scans are currently supported");
+    node_->get_parameter_or<std::string>("topic", topic, source);
```

Comment:
This is in the stream of inputs from configuring multiple sensor inputs (or types if some are to only clear or mark). I dont see here where `node_` or `source` are updated to go up and down into new sensor namespaces to get the parameters. Maybe I'm missing something.

Reply:
I think you are right, I overlooked the updated namespace of the parameters, however since only one node is being spun here, the parameters must exist on this node (or some other node also being spun for which we can create a client). I can add a prefix to the param names which will effectively do the same, though they will all live on the costmap node. This should be fine since yaml files support prefix namespaces on parameters. The only other solution I think would be to create (and spin) separate nodes for each source, which I don't believe we want to do.

Reply:
Yeah, that would be not good. The reason to have multiple sensors in the same node is so different sensors can clear out each other as the robot moves relative to obstacles (or obstacles themselves move). It's a pretty important feature that you can configure multiple marking/clearing sources in each layer.

Reply:
Yes, I understand the use case as you describe. I plan to update to prefix a parameter namespace as the source for the sensor.